hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java

```diff
@@ -151,25 +178,62 @@ public CallOutcome call() {
             final var callData = (hydratedEthTxData != null && hydratedEthTxData.ethTxData() != null)
                     ? Bytes.wrap(hydratedEthTxData.ethTxData().callData())
                     : null;
-            return CallOutcome.fromResultsWithMaybeSidecars(
+            final var outcome = CallOutcome.fromResultsWithMaybeSidecars(
                     result.asProtoResultOf(ethTxDataIfApplicable(), rootProxyWorldUpdater, callData),
                     result.asEvmTxResultOf(ethTxDataIfApplicable(), callData),
                     result.isSuccess() ? rootProxyWorldUpdater.getUpdatedContractNonces() : null,
                     result.isSuccess() ? rootProxyWorldUpdater.getCreatedContractIds() : null,
                     result.isSuccess() ? result.evmAddressIfCreatedIn(rootProxyWorldUpdater) : null,
                     result);
+
+            final var elapsedMs = System.nanoTime() - startTimeNanos;
+
+            // Update the ops duration throttle
+            if (opsDurationThrottleEnabled) {
+                throttleAdviser.consumeOpsDurationThrottleCapacity(opsDurationCounter.opsDurationUnitsConsumed());
+            }
+
+            recordProcessedTransactionToMetrics(
+                    hevmTransaction, outcome, elapsedMs, opsDurationCounter.opsDurationUnitsConsumed());
+
+            return outcome;
         } catch (HandleException e) {
             final var sender = rootProxyWorldUpdater.getHederaAccount(hevmTransaction.senderId());
             final var senderId = sender != null ? sender.hederaId() : hevmTransaction.senderId();
 
-            return maybeChargeFeesAndReturnOutcome(
+            final var outcome = maybeChargeFeesAndReturnOutcome(
                     hevmTransaction.withException(e),
                     senderId,
                     sender,
                     hevmTransaction.isContractCall() && contractsConfig.chargeGasOnEvmHandleException());
+
+            // Update the ops duration throttle
+            if (opsDurationThrottleEnabled) {
+                throttleAdviser.consumeOpsDurationThrottleCapacity(opsDurationCounter.opsDurationUnitsConsumed());
+            }
+
+            if (e.getStatus() == THROTTLED_AT_CONSENSUS) {
```

Comment:
❓ This is now unreachable code since we've switched to `EXCEPTIONAL_HALT` at the point of exhausting `opsDuration`, right?

Reply:
yes, this should be removed.